cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ProgramLifecycleService.java

```diff
@@ -497,10 +696,19 @@ public void run() {
         RunRecordsCorrectorRunnable.LOG.debug("End correcting invalid run records.");
       } catch (Throwable t) {
         // Ignore any exception thrown since this behaves like daemon thread.
+        //noinspection ThrowableResultOfMethodCallIgnored
         LOG.warn("Unable to complete correcting run records: {}", Throwables.getRootCause(t).getMessage());
         LOG.debug("Exception thrown when running run id cleaner.", t);
       }
     }
   }
 
+  private Program getProgram(ProgramId id)
+    throws IOException, ApplicationNotFoundException, ProgramNotFoundException {
+    Program program = store.loadProgram(id.toId());
+    if (program == null) {
```

Comment:
Hmm, ``DefaultStore`` marks ``loadProgram`` as ``@Nullable``, but doesn't return null. we can remove it, I think